build/NugetVersionDeprecator/Program.cs

```diff
@@ -0,0 +1,1016 @@
+                packageName,
+                includePrerelease: false,
+                includeUnlisted: false,
+                cache,
+                NullLogger.Instance,
+                CancellationToken.None)).Cast<PackageSearchMetadata>()
+            // because of how NuGet query works, we'll get a whole lot of "close" match results - so we have to narrow the response down
+            // to only those packages where the name matches packageName
+            .Where(p =>
+                p.DeprecationMetadata is null // not deprecated
+                && string.Equals(p.Identity.Id, packageName, StringComparison.CurrentCultureIgnoreCase)).ToList();
+
+        // get the nuget packages with versions matching the list of all agent releases
+        var currentVersions = agentReleases.Select(ar => NuGetVersion.Parse(ar.Version)).ToList();
+        var packagesToDeprecate = packages.Where(p => currentVersions.Contains(p.Version)).ToList();
+
+        return packagesToDeprecate.Select(p =>
+            new PackageDeprecationInfo() { PackageName = p.PackageId, PackageVersion = p.Version.ToString() });
+    }
+
+    static async Task CreateGhIssueAsync(string message, string githubToken)
+    {
+        var ghClient = new GitHubClient(new Octokit.ProductHeaderValue("NugetVersionDeprecator"));
+        var tokenAuth = new Credentials(githubToken);
+        ghClient.Credentials = tokenAuth;
+
+
+        var newIssue = new NewIssue($"chore(NugetDeprecator): Deprecate Nuget packages.")
+        {
+            Body = message
+        };
+
+        newIssue.Labels.Add("Deprecation");
+        newIssue.Labels.Add("Nuget");
+
+        var issue = await ghClient.Issue.Create("newrelic", "newrelic-dotnet-agent", newIssue);
+
+        Console.WriteLine($"Created new GitHub Issue #{issue.Id} with title {issue.Title}.");
+    }
+
+    static Configuration LoadConfiguration(string path)
+    {
+        var input = File.ReadAllText(path);
+        var deserializer = new YamlDotNet.Serialization.Deserializer();
+        return deserializer.Deserialize<Configuration>(input);
+    }
+
+    static void ValidateOptions(Options opts)
+    {
+        if (!opts.TestMode && string.IsNullOrEmpty(opts.GithubToken))
+        {
+            ExitWithError(ExitCode.BadArguments, "Github token is required when not in Test mode.");
+        }
+        if (!File.Exists(opts.ConfigurationPath))
+        {
+            ExitWithError(ExitCode.FileNotFound, $"Configuration file did not exist at {opts.ConfigurationPath}.");
+        }
+    }
+
+    static void HandleParseError(IEnumerable<Error> errs)
+    {
+        ExitWithError(ExitCode.BadArguments, "Error occurred while parsing command line arguments.");
+    }
+
+    static void ExitWithError(ExitCode exitCode, string message)
+    {
+        Console.WriteLine(message);
+        Environment.Exit((int)exitCode);
+    }
+
+    // for testing only
+    private const string NerdGraphSample = @"
```

Comment:
I think this can be removed or moved to a text file/other file to make the class cleaner.